packages/suite/src/actions/wallet/discoveryActions.ts

```diff
@@ -315,30 +333,24 @@ export const start = () => async (dispatch: Dispatch, getState: GetState): Promi
 
                 // add failed coins to discovery
                 dispatch(
-                    update({ device, failed, total: discovery.total - LIMIT * failed.length }),
+                    update({
+                        device,
+                        failed,
+                        total: discovery.total - LIMIT * failed.length,
+                        bundleSize: discovery.bundleSize - failed.length,
+                    }),
                 );
 
                 await dispatch(start()); // restart process, exclude failed coins
                 return;
             } catch (error) {
-                // do nothing. error will be handled later
+                // do nothing. error will be handled in lower block
```

Comment:
are we really sure we don't want to at least add console.log (if we don't want to notify user about it) so it is not silent fail?